src/lib.rs

```diff
@@ -123,6 +131,18 @@ impl<'a> PrivateKeyDer<'a> {
     }
 }
 
+#[cfg(feature = "alloc")]
+impl PemObject for PrivateKeyDer<'static> {
+    fn from_pem(kind: SectionKind, value: Vec<u8>) -> Option<Self> {
+        match kind {
```

Comment:
So to call this out specifically: should we validate here that the DER is in the correct format? If we don't, I suppose rustls is going to error out at a later stage?

Reply:
Yeah, this is maintaining the status-quo. We could have a `check_format()` or something on `PrivateKeyDer` (extracts the data, guesses the format, and checks that is consistent with the used variant) but it would come short of any sort of key validation or thorough validation of the format -- I think these things need to happen at the level that fully supports DER decoding and cryptography.

Reply:
> it would come short of any sort of key validation or thorough validation of the format -- I think these things need to happen at the level that fully supports DER decoding and cryptography.

Not sure I agree? The earlier we can signal errors, the more context might be available to the user, especially in this case of format confusion that I imagine would be tricky for inexperienced users. But, it is the status quo so no need to block this PR on it.